api_tests/comments/test_views.py

```diff
@@ -0,0 +1,1037 @@
+        self.public_url = '/{}comments/{}/'.format(API_BASE, self.public_comment._id)
+        self.public_comment_payload = {
+            'data': {
+                'id': self.public_comment._id,
+                'type': 'comments',
+                'attributes': {
+                    'content': 'Updating this comment',
+                    'deleted': False
+                }
+            }
+        }
+
+    def _set_up_registration_with_comment(self):
+        self.registration = RegistrationFactory(creator=self.user)
+        self.registration_comment = CommentFactory(node=self.registration, user=self.user)
+        self.registration_url = '/{}comments/{}/'.format(API_BASE, self.registration_comment._id)
+
+    def test_private_node_logged_in_contributor_can_view_comment(self):
+        self._set_up_private_project_with_comment()
+        res = self.app.get(self.private_url, auth=self.user.auth)
+        assert_equal(res.status_code, 200)
+        assert_equal(self.comment._id, res.json['data']['id'])
+
+    def test_private_node_logged_in_non_contributor_cannot_view_comment(self):
+        self._set_up_private_project_with_comment()
+        res = self.app.get(self.private_url, auth=self.non_contributor.auth, expect_errors=True)
+        assert_equal(res.status_code, 403)
+
+    def test_private_node_logged_out_user_cannot_view_comment(self):
+        self._set_up_private_project_with_comment()
+        res = self.app.get(self.private_url, expect_errors=True)
+        assert_equal(res.status_code, 401)
+
+    def test_public_node_logged_in_contributor_can_view_comment(self):
+        self._set_up_public_project_with_comment()
+        res = self.app.get(self.public_url, auth=self.user.auth)
+        assert_equal(res.status_code, 200)
+        assert_equal(self.public_comment._id, res.json['data']['id'])
+
+    def test_public_node_logged_in_non_contributor_can_view_comment(self):
+        self._set_up_public_project_with_comment()
+        res = self.app.get(self.public_url, auth=self.non_contributor.auth)
+        assert_equal(res.status_code, 200)
+        assert_equal(self.public_comment._id, res.json['data']['id'])
+
+    def test_public_node_logged_out_user_can_view_comment(self):
+        self._set_up_public_project_with_comment()
+        res = self.app.get(self.public_url)
+        assert_equal(res.status_code, 200)
+        assert_equal(self.public_comment._id, res.json['data']['id'])
+
+    def test_registration_logged_in_contributor_can_view_comment(self):
+        self._set_up_registration_with_comment()
+        res = self.app.get(self.registration_url, auth=self.user.auth)
+        assert_equal(res.status_code, 200)
+        assert_equal(self.registration_comment._id, res.json['data']['id'])
+
+    def test_comment_has_user_link(self):
+        self._set_up_public_project_with_comment()
+        res = self.app.get(self.public_url)
+        url = res.json['data']['relationships']['user']['links']['related']['href']
```

Comment:
It's generally helpful to check the return code too; if the request fails the error will be a lot more readable. (for this and several tests below)